src/__experimental__/components/radio-button/radio-button.component.js

```diff
@@ -82,7 +82,15 @@ RadioButton.propTypes = {
    * The name of the group containing the RadioButton (can also be set via
    * the 'groupName' prop of the RadioButtonGroup component)
    */
```

Comment:
Not sure if that's enough, you can still pass the name prop to the RadioButton which gets passed on to CheckableInput even though it warns the user about not doing so

Reply:
Sure, you can pass the prop but you will get the console error that you shouldn't do it. The same situation is in other libraries

Reply:
I agree with @mkrds we should discard the property AND warn about it.

Reply:
I'm not big fan of the solution but I lost 2:1 :D 
Updated in https://github.com/Sage/carbon/pull/2389/commits/41792bfa5183738d2d45bf3ad66fe318edb8df1a

Reply:
Yeah, delete is very slow. Can you not refactor it so we don't have three names for the same thing (`name`, `groupName`, `inputName`). Really we need to rename the `inputName` prop to be `name`?

Reply:
Not a fan of that solution either, but I guess that thanks to that we can free users of some hard to debug bugs. Anyway I'd rather not mutate the prop object and insted of deleting one property I'd just desctructure it like that:
`
const RadioButton = ({
  id, label, onChange, value, name, ...props
})
`

Reply:
@mkrds that fails eslint rule, `no-unused-variables`. Why do you not want to rename `inputName`?
The intended interface is `<RadioGroup name="foobar"><RadioButton /></RadioGroup>`